src/typescript/11-keyof-typeof/keyof-typeof.md

```diff
@@ -115,28 +119,84 @@ const getDinoFacts = () => {
   };
 };
 
-/**
- *{name: string, size: {weight: {amount: number; unit: string;}}, facts: string[]}
- */
 type DinoFacts = ReturnType<typeof getDinoFacts>;
 ```
+@highlight 12
+
+For the above example, given the `typeof DinoFacts`, the `ReturnType` is now: 
+`{name: string, size: {weight: {amount: number; unit: string;}}, facts: string[]}`
+
+There are two `typeof`s to be aware of in TypeScript – the JavaScript one and the TypeScript one. The difference between them is the context in which they are used.
 
-There are two `typeof`s to be aware of in TypeScript – the JavaScript one and the TypeScript one. The difference between them is the context in which they are used. If it is being used in an expression context, (used as a part of your code) it is the JavaScript `typeof` and will return a string with one of nine JavaScript types. And if it’s used in a type context (as part of your type declarations) it’s the TypeScript `typeof` and will refer to the type of whatever values follow it.
+If it is being used in an expression context, (used as a part of your code) it is the JavaScript `typeof` and will return a string with one of nine JavaScript types.
+
+If it’s used in a type context (as part of your type declarations) it’s the TypeScript `typeof` and will refer to the type of whatever values follow it.
```

Comment:
A suggestion for how to make this section more clear:

In TypeScript, you'll encounter two different usages of `typeof`—one from JavaScript and another specific to TypeScript. Understanding when and how each is used is crucial for effective type management and code clarity.

**JavaScript `typeof`:** This version of `typeof` is used within code expressions. When you apply `typeof` to a variable, TypeScript treats it as the JavaScript operator, returning a string that represents the variable’s primitive type.

For example, `typeof "hello"` results in `"string"`, and `typeof 42` results in `"number"`. This is consistent with JavaScript’s behavior, where typeof is typically used to determine the type of a runtime value.

**TypeScript `typeof`:** When `typeof` is used in a type context—specifically, in type declarations or annotations—it behaves differently. Here, `typeof` is used to capture and use the type of a variable rather than its value.

For instance, if you have `const x = "hello"`, using `type Y = typeof x` in TypeScript will set Y as the type `"string"`, effectively using the type of `x` rather than its value.

It’s important to note that typeof in TypeScript, particularly in type contexts, has its limitations. You can only use typeof on identifiers (such as variable names) and their properties. Attempting to use typeof on more complex expressions or certain values directly will not work as expected and can lead to errors.